### PR TITLE
Embedded daemon in one unified binary

### PR DESCRIPTION
**Pull request by marbetschar**
This PR fixes the notification icon issue from #677 by embedding the daemonized background process into the same application binary (in the screenshot: "Other" is before this change, the correct notification is after this change):

Screenshot from 2021-09-08 08-52-28

This way, we gain the following benefits:

### 1. The notification icon is correct

The notification icon is read from the `Icon` key stored in `/usr/share/applications/{application_id}.desktop`. Since the background process shares the same `application_id` as the main application, the notification server uses the correct icon to display the alert.

### 2. The background process starts automatically

Autostart files are read from `/etc/xdg/autostart/{application_id}.desktop`. We can easily place a second desktop file with the EXEC line `io.elementary.calendar --background` in this directory.

### 3. When clicking on the notification, the corresponding app opens

When clicking on the notification, the notification server executes the `Exec` key of the `/usr/share/applications/{application_id}.desktop` file. Here it is crucial we embedded the background process within the same binary - because this way, the regular binary starts and the user is presented with the app itself. Just like we want it.

### 4. `System Settings > Notifications` work as expected

[As it turned out when working on this in Mail](https://github.com/elementary/mail/pull/679#issuecomment-914469849), if we don't embed the background process within the same binary we either end up with two application entries there - or even worse, with one which does not have any effect at all.

### CODE REVIEW

File: src/TodayEventMonitor.vala
Comment (mcclurgm) [inline src/TodayEventMonitor.vala:85]: This probably makes more sense as an if statement? There's currently only one case, so it would make sense that way as an if/else. But we can leave it as is to make future action types easier to add, if you want.

File: src/TodayEventMonitor.vala
Comment (mcclurgm) [inline src/TodayEventMonitor.vala:113]: Since the alarm has an action that we're not parsing, this would probably be a good case for a warning.

File: src/TodayEventMonitor.vala
Comment (mcclurgm) [inline src/TodayEventMonitor.vala:133]:
```suggestion
    public void send_event_notification (ECal.Component event, string uid) {
```

I think this is a little clearer. The `add_timeout` function is really "queuing" the notification in the sense of adding it to a queue, while this function is called to actually send the notification.

File: src/TodayEventMonitor.vala
Comment (mcclurgm) [inline src/TodayEventMonitor.vala:57]: Rather strangely, I'm seeing that when I edit an event the `events` collection has size 0, so the callback is never added. I'm not sure quite how to debug this, so if you have any suggestions, let me know. In the meantime I'll keep digging and update if I find anything.

**Review by mcclurgm — CHANGES_REQUESTED** [review]
I have some very small changes I noted. There's a bigger issue here that I noted for `on_events_updated`, but I'll come back with a second review when I can figure out what's going on there.

File: src/TodayEventMonitor.vala
Comment (mcclurgm) [reply src/TodayEventMonitor.vala:57]: Never mind, I was seeing a combination of an EventStore bug and a user error about notifications. Ignore this.

**Comment by mcclurgm** [comment]
Seems to work as expected. As I said in my last comment there are a few minor style things, but I think this pretty much works! I'll approve it once we get those sorted.

File: src/TodayEventMonitor.vala
Comment (marbetschar) [reply src/TodayEventMonitor.vala]: Yeah, an if makes more sense here.

File: src/TodayEventMonitor.vala
Comment (marbetschar) [reply src/TodayEventMonitor.vala]: Added a warning.

File: src/TodayEventMonitor.vala
Comment (marbetschar) [reply src/TodayEventMonitor.vala]: Renamed the function.

**Comment by marbetschar** [comment]
@mcclurgm thanks for your review! I addressed the proposed changes and I think we should be good to go here now.

**Comment by mcclurgm** [comment]
Alright, seems good to me.

**Comment by ghsttwn** [comment]
Latest Calendar version comes with the autostart entry for the daemon (at `/etc/xdg/autostart/io.elementary.calendar-daemon.desktop`) though it never really *starts* the daemon. 😕 Was this intended?

**Comment by marbetschar** [comment]
@ghsttwn no, this was not the intention :)

The daemon should be autostarted as soon as you end your session and start a new one (read "logout and login again"). Doesn't this work on your end? If not, can you please open a new issue with a link to this PR, so we can investigate there?

**Pull request merged**
